app/models/route.rb

```diff
@@ -1,7 +1,19 @@
 class Route < ApplicationRecord
   validates :name, presence: true
 
-  has_many :railway_stations_routes, dependent: :nullify
-  has_many :railway_stations, through: :railway_stations_routes
+  has_many :railway_stations_routes, dependent: :delete_all
+  has_many :railway_stations, -> { order('position') }, through: :railway_stations_routes
   has_and_belongs_to_many :trains
+
+  before_validation :set_name, :stations_count_validation
+
+  private
+
+  def set_name
+    self.name = "#{railway_stations.first.name} - #{railway_stations.last.name}"
```

Comment:
I think better use https://zaiste.net/rails_activerecord_updating_attributes_object_fields/ instead of self.